nix-buffer.el

```diff
@@ -216,9 +216,10 @@ is removed."
   (interactive)
   (let* ((root (directory-file-name (or (buffer-file-name) default-directory)))
 	 (expr-dir (locate-dominating-file root "dir-locals.nix")))
-    (when expr-dir
-	 (let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
-	   (nix-buffer--nix-build root expr-file)))))
+    (if expr-dir
+	(let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
+	  (nix-buffer--nix-build root expr-file))
+      (error "No dir-locals.nix file found"))))
```

Comment:
Okay, I was wrong on ```projectile-mode-hook```. I thought it would run each time projectile-mode was used in a buffer. Instead, I think it just runs once so you'll need some kind of hook like  nix-buffer any way. Does ```find-file-hook``` get run in comint or eshell as well? Another hook that is similar is ```after-change-major-mode-hook```.